airflow/models/__init__.py

```diff
@@ -2872,6 +2885,143 @@ def create_dagrun(self,
                                             conf=conf,
                                             session=session)
 
+    def date_range(self, start_date, num=None, end_date=utcnow()):
+        if num:
+            end_date = None
+        return utils_date_range(
+            start_date=start_date, end_date=end_date,
+            num=num, delta=self._schedule_interval)
+
+    def is_fixed_time_schedule(self):
+        """
+        Figures out if the DAG schedule has a fixed time (e.g. 3 AM).
+
+        :return: True if the schedule has a fixed time, False if not.
+        """
+
+        # TODO: Fix behaviour on timedelta's.
+
+        now = datetime.now()
+        cron = croniter(self._schedule_interval, now)
+
+        start = cron.get_next(datetime)
+        cron_next = cron.get_next(datetime)
+
+        if cron_next.minute == start.minute and cron_next.hour == start.hour:
```

Comment:
This could be simplified to `return cron_next.minute == start.minute and cron_next.hour == start.hour`. Less is more :-)